pkg/controller/jobs/rayjob/rayjob_controller_test.go

```diff
@@ -268,3 +268,78 @@ func TestNodeSelectors(t *testing.T) {
 		})
 	}
 }
+
+func Test_RayJobFinished(t *testing.T) {
+	testcases := []struct {
+		name             string
+		status           rayv1.RayJobStatus
+		expectedSuccess  bool
+		expectedFinished bool
+	}{
+		{
+			name: "jobStatus=Running, jobDeploymentStatus=Running",
+			status: rayv1.RayJobStatus{
+				JobStatus:           rayv1.JobStatusRunning,
+				JobDeploymentStatus: rayv1.JobDeploymentStatusRunning,
+			},
+			expectedSuccess:  false,
+			expectedFinished: false,
+		},
+		{
+			name: "jobStatus=Succeeded, jobDeploymentStatus=Complete",
+			status: rayv1.RayJobStatus{
+				JobStatus:           rayv1.JobStatusSucceeded,
+				JobDeploymentStatus: rayv1.JobDeploymentStatusComplete,
+			},
+			expectedSuccess:  true,
+			expectedFinished: true,
+		},
+		{
+			name: "jobStatus=Failed, jobDeploymentStatus=Complete",
+			status: rayv1.RayJobStatus{
+				JobStatus:           rayv1.JobStatusFailed,
+				JobDeploymentStatus: rayv1.JobDeploymentStatusComplete,
+			},
+			expectedSuccess:  false,
+			expectedFinished: true,
+		},
+		{
+			name: "jobStatus=Failed, jobDeploymentStatus=Failed",
+			status: rayv1.RayJobStatus{
+				JobStatus:           rayv1.JobStatusFailed,
+				JobDeploymentStatus: rayv1.JobDeploymentStatusFailed,
+			},
+			expectedSuccess:  false,
+			expectedFinished: true,
+		},
+		{
+			name: "jobStatus=Running, jobDeploymentStatus=Failed (when activeDeadlineSeconds is exceeded)",
```

Comment:
I'm good with this workaround in Kueue, but wondering if we should additionally ticket this in Ray (I would expect jobStatus=Failed in this case).